webapi/src/test/scala/org/knora/webapi/slice/resources/repo/service/value/queries/CreateValueQueryBuilderSpec.scala

```diff
@@ -0,0 +1,1208 @@
+        comment = None,
+      )
+  }
+
+  // Helper to compare SPARQL queries for strict equivalence
+  def compareSparqlQueries(twirlQuery: String, builderQuery: String) =
+    for {
+      parsedTwirlUpdate   <- ZIO.attempt(UpdateFactory.create(twirlQuery))
+      parsedBuilderUpdate <- ZIO.attempt(UpdateFactory.create(builderQuery))
+      normalizedTwirl      = replaceUuidPatterns(parsedTwirlUpdate.toString)
+      normalizedBuilder    = replaceUuidPatterns(parsedBuilderUpdate.toString)
+      assertEqual          = assertTrue(normalizedBuilder == normalizedTwirl)
+    } yield assertEqual
+
+  def replaceUuidPatterns(sparqlQuery: String): String = {
+    val uuidPattern = """knora-base:valueHasUUID\s+"[^"]+"\s*[.;]""".r
+    uuidPattern.replaceAllIn(sparqlQuery, """knora-base:valueHasUUID "0000000000000000000000" .""")
+  }
+}
+
+object CreateValueQueryBuilderSpec extends ZIOSpecDefault with GoldenTest {
+  val rewriteAll: Boolean = false
+
+  import CreateValueQueryBuilderTestSupport.*
+
+  private def validateLiteralContains(
+    parsedQuery: UpdateRequest,
+    predicate: String,
+    validator: String => Boolean,
+    errorMessage: String,
+  ): Task[Unit] =
+    ZIO.attempt {
+      parsedQuery.getOperations.forEach(_.visit(new UpdateVisitorBase {
+        override def visit(update: UpdateModify): Unit =
+          update
+            .getInsertQuads()
+            .asScala
+            .toList
+            .find(_.getPredicate.toString == predicate)
+            .fold(throw new AssertionError("Expected property not found")) { quad =>
+              val value = quad.getObject.getLiteralValue.toString
+              if (!validator(value)) {
+                throw new AssertionError(errorMessage)
+              }
+            }
+      }))
+    }
+
+  private def validateUriContains(
+    parsedQuery: UpdateRequest,
+    validator: String => Boolean,
+    errorMessage: String,
+  ): Task[Unit] =
+    ZIO.attempt {
+      parsedQuery.getOperations.forEach(_.visit(new UpdateVisitorBase {
+        override def visit(update: UpdateModify): Unit =
+          update
+            .getInsertQuads()
+            .asScala
+            .toList
+            .find(_.getPredicate.toString == OntologyConstants.KnoraBase.ValueHasUri)
+            .fold(throw new AssertionError("Expected property not found")) { quad =>
+              val value = quad.getObject.toString
+              if (!validator(value)) {
+                throw new AssertionError(errorMessage)
+              }
+            }
+      }))
+    }
+
+  // Comparison test suites per value type
+
+  def spec = suite("CreateValueQueryBuilderSpec")(
+    suite("Builder vs Twirl template comparison")(
+      suite("TextValueContentV2")(
+        test("without comment or language") {
+          val testValue = TestDataFactory.createTextValue()
+          for {
+            twirlQuery   <- ZIO.attempt(TestDataFactory.createTwirlQuery(testValue))
```

Comment:
ok, I see, the twirl stuff is still in use. But maybe it would be better to change these tests to the golden testing thing too?